### PR TITLE
3187 enhance test clients for eth tx

### LINKED ISSUE #3187: Enrich testclients to be able to run eth tx e2e tests
We need to update our test framework in order to set the foundation to:
- add e2e tests for eth tx
- update the existing e2e with such calls

### PR DESCRIPTION
**Pull request by IvanKavaldzhiev**
**Description**:
In order to make E2E tests to support `HIP-410` we would need a new type of operations in the `test-clients` framework, so that we send `Ethereum` wrapped transactions to the `hedera-nodes`. 

This PR will add the needed logic for the test-clients to become `Ethereum` compliant.

**Related issue(s)**:

Fixes #3187 

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)
- [x] Introducing new type of HapiTxnOp - `HapiEthereumCall` for executing wrapped `Ethereum` calls in a `Hedera` transaction 
- [x] Introducing new type of HapiTxnOp - `HapiEthereumContractCreate` for executing wrapped `Ethereum` contract creates in a `Hedera` transaction
- [x] Add `HelloWorldEthereumSuite` for making PoC E2E tests with `Ethereum` operations
- [x] (GItHub Actions) - Add separate executing nodes for execution of existing E2E tests with replaced `contract calls` with `Ethereum calls`
- [x] Adding possibility to run locally an E2E Suite with both `contract` and `Ethereum` calls (double execution of the same context with replaced calls)

### CODE REVIEW

File: hedera-node/data/config/api-permission.properties [hunk 1]
```diff
@@ -0,0 +1,68 @@
+# Crypto
```
Comment (georgiyazovaliiski) [inline hedera-node/data/config/api-permission.properties:1]: I think this file is git-ignored. Is there any particular reason that it is still committed anyway?

File: hedera-node/src/main/java/com/hedera/services/txns/ethereum/EthereumTransitionLogic.java [hunk 2]
```diff
@@ -194,12 +194,15 @@ public Function<TransactionBody, ResponseCodeEnum> semanticCheck() {
 	public void preFetch(final TxnAccessor accessor) {
 		var ethTxData = spanMapAccessor.getEthTxDataMeta(accessor);
 
-		EthereumTransactionBody op = accessor.getTxn().getEthereumTransaction();
-		ethTxData = maybeUpdateCallData(accessor, ethTxData, op);
+//		EthereumTransactionBody op = accessor.getTxn().getEthereumTransaction();
+//		ethTxData = maybeUpdateCallData(accessor, ethTxData, op);
```
Comment (georgiyazovaliiski) [inline hedera-node/src/main/java/com/hedera/services/txns/ethereum/EthereumTransitionLogic.java:198]: Were those two left to be like that on purpose even though they have been moved a couple of lines down?

File: test-clients/src/main/java/com/hedera/services/bdd/spec/utilops/EthTxData.java [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+                gasLimit,
+                to,
+                value,
+                callData,
+                accessList,
+                recId,
+                v,
+                r,
+                s);
+    }
```
Comment (georgiyazovaliiski) [inline test-clients/src/main/java/com/hedera/services/bdd/spec/utilops/EthTxData.java:147]: You might want to wrap this with a try/catch clause and return null on invalid parsing, similarly to what is currently implemented in the `populateEthTxData` in the `eth-tx-interop` branch.

File: test-clients/src/main/java/com/hedera/services/bdd/spec/utilops/EthTxDataBuilder.java [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+import com.esaulpaugh.headlong.util.Integers;
+import com.sun.jna.ptr.IntByReference;
+import org.bouncycastle.jcajce.provider.digest.Keccak;
+
+import java.math.BigInteger;
+import java.nio.ByteBuffer;
+import java.util.List;
+
+public class EthTxDataBuilder {
+
```
Comment (georgiyazovaliiski) [inline test-clients/src/main/java/com/hedera/services/bdd/spec/utilops/EthTxDataBuilder.java:13]: nit: whitespace.

File: test-clients/src/main/java/com/hedera/services/bdd/spec/utilops/UtilStateChange.java [hunk 5]
```diff
@@ -55,4 +69,30 @@ public static List<ContractStateChange> stateChangesToGrpc(List<StateChange> sta
...
+	public static void markSpecAsBeenExecuted(final String spec) {
+		specToBeenExecuted.putIfAbsent(spec, true);
+	}
+
+	public static boolean hasSpecBeenExecuted(final String spec) {
+		if(specToBeenExecuted.containsKey(spec)) {
+			return specToBeenExecuted.get(spec);
+		} else {
+			return false;
+		}
```
Comment (georgiyazovaliiski) [inline test-clients/src/main/java/com/hedera/services/bdd/spec/utilops/UtilStateChange.java:96]: nit: can remove the else block and directly return false after the if has been skipped.

File: hedera-node/src/main/java/com/hedera/services/store/contracts/precompile/HTSPrecompiledContract.java [hunk 6]
```diff
@@ -366,13 +366,7 @@ void prepareFields(final MessageFrame frame) {
...
 		this.sideEffectsTracker = sideEffectsFactory.get();
 		this.ledgers = updater.wrappedTrackingLedgers(sideEffectsTracker);
-
-		final var unaliasedSenderAddress = updater.unaliased(frame.getSenderAddress().toArray());
-		if (unaliasedSenderAddress != null) {
-			this.senderAddress = Address.wrap(Bytes.of(unaliasedSenderAddress));
-		} else {
-			this.senderAddress = frame.getSenderAddress();
-		}
+		this.senderAddress = frame.getSenderAddress();
```
[the remark this answers is not in the document]
Comment (tinker-michaelj) [reply hedera-node/src/main/java/com/hedera/services/store/contracts/precompile/HTSPrecompiledContract.java:369]: @IvanKavaldzhiev I'm pretty sure we **do** need to un-alias the `sender` for the ERC-20/721 `transfer`s to work. Imo the solution is to use a separate helper method to un-alias this address specifically, one that doesn't throw `INVALID_SOLIDITY_ADDRESS` when the mirror form is used.

I did so in [this PR](https://github.com/hashgraph/hedera-services/pull/3286).

**Pull request merged**